### ISSUE TITLE
Ability to generate lineloads along different axis

### ISSUE THREAD

**Issue by khalilsb** [nasa/cape#29]
Currently lineloads are defaulted to be along the x direction. It would be beneficial to make this an input in the case of extracting lineloads in other directions for example for a wing in the spanwise direction

**Comment by nasa-ddalle** [comment]
Yes, this is a great idea. Definitely `triload` can support that (and also using an arbitrary vector to define the cut planes), but let me see how quickly this can be integrated into CAPE. It's definitely worth doing, and I'm already planning an overhaul of the line load database code. But it might be something easy (at least to set it to `y` or `z`) that can be done before the overhaul.

**Comment by nasa-ddalle** [comment]
I think this will work for now if you set `"CutPlaneNormal"` to `"y"` in the `"DataBook"` definition for your component. It will look kind of weird in the DataBook because it will still refer to the coordinate as `"x"` in the output files, but you should be able to verify that it's actually doing what you want by looking at the `.triq` files or other output files created by `triloadCmd` in the `lineload/` folder within an individual case.

**Comment by khalilsb** [comment]
thanks for the update! I gave this a try with the latest version of cape and was running into an issue where cape is not running triloads due to not having enough iterations, this seems to be different behavior than cape v1.0. I thought lineloads were extracted from the latest boundary*plt

```
        // Line load
        "ll_wing": {
            "Type": "LineLoad",
            "CompID": ["wing_py","wingtip_py"],
            "CutPlaneNormal": "y",
            "nCut": 200
        },    
```

Here is the output from cape 
```
    pyfun --ll

    oml192_rev0_de_py+00_de_ny+00_ds_py+00_ds_ny+00_dr+15/m0.30a21.0b0.0
      Not enough iterations (None) for analysis.
```

**Comment by nasa-ddalle** [comment]
Took me a while to figure out what was happening here ... The commits on this page

https://github.com/nasa/cape/commits/main/cape/pyfun/case.py

b/w 3/22 and 3/25 should fix this issue. CAPE was finding the `{PROJECT}_tec_boundary.plt` file but wasn't able to figure out what iteration it came from. Now it should be able to do that and also be smarter about a few other metadata situations. I've only updated this function on the `main` branch; back-porting it to `v1.1` would be some work.

**Comment by khalils2** [comment]
Just recently updated cape to v1.2 and it looks like SectionType is longer a lineload input. Can this be added back?

**Comment by nasa-ddalle** [comment]
I'm curious if there's a particular setup that's not working? My testing still has `"SectionType"` in v1.2... Maybe there's a subtler problem with the way your JSON file is being read? I think there was a small change to how the options are read, so I might have made a mistake there.

**Comment by khalils2** [comment]
As i was writing this up i found the typo that caused the error! Apologies for the false alarm 
![error](https://github.com/nasa/cape/assets/15007986/825eb7c0-968c-4f21-a16c-d3dbb8ea51f6)

But since upgrading i was getting some odd warnings regarding numpy, not sure what has changed here, might be something on my end

**Comment by nasa-ddalle** [comment]
Ah! A hard-to-see typo.

It looks like there's a NumPy 2.0 now ... and I might have to rewrite the extension a bit. For now, it might be easiest to just use the `py3-none-any` version. You won't lose much unless you are manipulating surface grids a lot with CAPE.

**Comment by khalils2** [comment]
is it easy to add in a feature to specify the max and min location of the lineload definition? I know this is a triload input but cape defaults to extracting the lineload on the entire component extents

**Comment by nasa-ddalle** [comment]
No, I don't think there is! I should be able to add this pretty easily, and the release of CAPE 1.2.2 should also come out shortly... It'll help with that NumPy 2.0 thing as well.

**Comment by khalils2** [comment]
Hi Derek, i saw cape 1.2.2 was released last week, is it compatible with numpy 2.0?

**Comment by nasa-ddalle** [comment]
Yes, as far as I can tell! The main thing has to do with how NumPy 2.0 has outputs like `np.int32(3)` instead of just `3`, and that's been taken care of. The other thing is the extension modules ... I decided to have it compile against NumPy 2.0+ if your Python version is 3.10 or newer and against the older API for older Python versions.

**Comment by khalils2** [comment]
Thanks for getting back to me. i was able to get cape 1.2.2 installed on python3.10 and the initial _ARRAY_API errors seems to be fixed and cape is setting up all the nml files correctly but when i go to run a case i still seem to get an error that looks to be slurm related? Also i noticed cape is not creating a symbolic link for the fun3d.nml file, not sure if it just has not gotten to the step yet 

![image](https://github.com/user-attachments/assets/6c735d5f-1de5-41ea-8d07-2be9e41b23a4)

**Comment by nasa-ddalle** [comment]
Yeah, that should say `os.environ.get("SLURM_NTASKS", "1")`

I can push an update as well, but you should be pretty safe to fix that one line in your installed version. The `fun3d.nml` symlink happens just before starting `nodet`, so it makes sense that didn't get created.

**Comment by khalils2** [comment]
I think we may have found a bug in the latest CAPE version 1.2.2, it seems like the CAPE is excluding the last component in the mapbc file in all the component definition in the nml, I have tried it with both version and its fine in the older version. not sure what would cause this could it be related the numpy version change?

**Comment by nasa-ddalle** [comment]
You're saying that you might have something like
```
component_input(7) = '7-8,10-13'
```
and you really want it to be
```
component_input(8) = '7-8,10-14'
```
(and that your `.mapbc` file has 14 entries). Is that right? I'm not seeing that behavior in any of my versions, so I just want to clarify if I'm understanding the problem correctly.

**Comment by khalils2** [comment]
that is correct, this is the behavior i am seeing, we are still using the xml files to define the face ID numbers, not sure if that plays into it

**Comment by nasa-ddalle** [comment]
Can you share a `.mapbc` file with me and just the `"Config"` section of your JSON? I haven't been able to observe this problem yet ... It's possible I just published a wheel at a particular commit that was bad, but I haven't deduced the problem yet.

**Comment by nasa-ddalle** [comment]
Ok, this should be fixed on main in 738cac5da2fa26b4b8880ed1d56a7bdb98aaf60f and on v1.2 in 529a590e46194f4d2efb7b7a17c2170cdcf752ba. There were some additional constraints needed for this bug to appear, which seems to be why I was too dumb to find it!
